### PR TITLE
Switch to torch.compile for dropout for torch v>2.2

### LINKED ISSUE #597: Performance regression of `bias_dropout_add_fused_train` after switching to `torch.compile`
There is a performance regression of `bias_dropout_add_fused_train()` after switching to `torch.compile` for jit fusion in this [PR](https://github.com/NVIDIA/TransformerEngine/pull/589).

Here are my profiling results. As we can see, from either host or device perspective, there's a big regression.

- torch.compile
image

- nvFuser
image

My profiling scripts:
```python
import torch
import transformer_engine as te

def profile_bias_dropout_add_fused_train(
     hidden_size: int, dtype: torch.dtype, seq_length: int, micro_batch_size: int
) -> None:
    # warm up
    te.pytorch.jit.warmup_jit_bias_dropout_add(hidden_size, dtype, seq_length, micro_batch_size)

    inp = torch.rand(
        (seq_length, micro_batch_size, hidden_size), dtype=dtype, device="cuda"
    )
    residual = torch.rand(
        (seq_length, micro_batch_size, hidden_size), dtype=dtype, device="cuda"
    )
    bias = torch.rand((hidden_size), dtype=dtype, device="cuda")
    dropout_rate = 0.1

    torch.cuda.profiler.start()
    for _ in range(10):
        torch.cuda.nvtx.range_push("fused bda")
        output = te.pytorch.jit.bias_dropout_add_fused_train(inp, bias, residual, dropout_rate)
        torch.cuda.nvtx.range_pop()
    torch.cuda.profiler.stop()

te.pytorch.jit.set_jit_fusion_options()
profile_bias_dropout_add_fused_train(1024, torch.float32, 1024, 2)
```

cc @ksivaman @minitu

### PR DESCRIPTION
**Pull request by ksivaman**
Fixes #597

### CODE REVIEW

**Review by timmoon10 — APPROVED** [review]
LGTM
